{
    "expand": "operations,versionedRepresentations,editmeta,changelog,renderedFields",
    "fields": {
        "aggregateprogress": {
            "percent": 0,
            "progress": 0,
            "total": 86400
        },
        "aggregatetimeestimate": 86400,
        "aggregatetimeoriginalestimate": 86400,
        "aggregatetimespent": null,
        "assignee": {
            "active": true,
            "avatarUrls": {
                "16x16": "https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=ashutosh.bapat&avatarId=36785",
                "24x24": "https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=ashutosh.bapat&avatarId=36785",
                "32x32": "https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=ashutosh.bapat&avatarId=36785",
                "48x48": "https://issues.apache.org/jira/secure/useravatar?ownerId=ashutosh.bapat&avatarId=36785"
            },
            "displayName": "Ashutosh Bapat",
            "key": "ashutosh.bapat",
            "name": "ashutosh.bapat",
            "self": "https://issues.apache.org/jira/rest/api/2/user?username=ashutosh.bapat",
            "timeZone": "Asia/Kolkata"
        },
        "components": [{
            "id": "12317406",
            "name": "Tests",
            "self": "https://issues.apache.org/jira/rest/api/2/component/12317406"
        }],
        "created": "2018-11-21T05:48:51.000+0000",
        "creator": {
            "active": true,
            "avatarUrls": {
                "16x16": "https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=ashutosh.bapat&avatarId=36785",
                "24x24": "https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=ashutosh.bapat&avatarId=36785",
                "32x32": "https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=ashutosh.bapat&avatarId=36785",
                "48x48": "https://issues.apache.org/jira/secure/useravatar?ownerId=ashutosh.bapat&avatarId=36785"
            },
            "displayName": "Ashutosh Bapat",
            "key": "ashutosh.bapat",
            "name": "ashutosh.bapat",
            "self": "https://issues.apache.org/jira/rest/api/2/user?username=ashutosh.bapat",
            "timeZone": "Asia/Kolkata"
        },
        "customfield_10010": null,
        "customfield_12310191": null,
        "customfield_12310192": null,
        "customfield_12310220": "2018-11-21T08:32:05.143+0000",
        "customfield_12310222": "1_*:*_1_*:*_10252671_*|*_3_*:*_4_*:*_541008_*|*_5_*:*_1_*:*_0_*|*_10002_*:*_5_*:*_1048489691",
        "customfield_12310230": null,
        "customfield_12310250": null,
        "customfield_12310290": null,
        "customfield_12310291": null,
        "customfield_12310300": null,
        "customfield_12310310": "6.0",
        "customfield_12310320": [{
            "archived": false,
            "id": "12343343",
            "name": "4.0.0",
            "released": false,
            "self": "https://issues.apache.org/jira/rest/api/2/version/12343343"
        }],
        "customfield_12310420": "9223372036854775807",
        "customfield_12310920": "9223372036854775807",
        "customfield_12310921": null,
        "customfield_12311020": null,
        "customfield_12311024": null,
        "customfield_12311120": null,
        "customfield_12311820": "0|s00q2o:",
        "customfield_12312022": null,
        "customfield_12312023": null,
        "customfield_12312024": null,
        "customfield_12312026": null,
        "customfield_12312220": null,
        "customfield_12312320": null,
        "customfield_12312321": null,
        "customfield_12312322": null,
        "customfield_12312323": null,
        "customfield_12312324": null,
        "customfield_12312325": null,
        "customfield_12312326": null,
        "customfield_12312327": null,
        "customfield_12312328": null,
        "customfield_12312329": null,
        "customfield_12312330": null,
        "customfield_12312331": null,
        "customfield_12312332": null,
        "customfield_12312333": null,
        "customfield_12312334": null,
        "customfield_12312335": null,
        "customfield_12312336": null,
        "customfield_12312337": null,
        "customfield_12312338": null,
        "customfield_12312339": null,
        "customfield_12312340": null,
        "customfield_12312341": null,
        "customfield_12312520": null,
        "customfield_12312521": "Tue Dec 04 04:31:14 UTC 2018",
        "customfield_12312720": null,
        "customfield_12312823": null,
        "customfield_12312920": null,
        "customfield_12312921": null,
        "customfield_12312923": null,
        "customfield_12313422": "false",
        "customfield_12313520": null,
        "description": "The testcase is intended to test REPL LOAD with retry. The test creates a partitioned table and a function in the source database and loads those to the replica. The first attempt to load a dump is intended to fail while loading one of the partitions. Based on the order in which the objects get loaded, if the function is queued after the table, it will not be available in replica after the load failure. But if it's queued before the table, it will be available in replica even after the load failure. The test assumes the later case, which may not be true always.\r\n\r\nHence fix the testcase to order the objects by a fixed ordering. By setting hive.in.repl.test.files.sorted to true, the objects are ordered by the directory names. This ordering is available with minimal changes for testing, hence we use it. With this ordering a function gets loaded before a table. So changed the test to not expect the function to be available after the failed load, but be available after the retry.\r\n\r\nWhile writing that testcase, I found that even if a function fails to load, it's visible through show functions and also is available to be called just as if the failure has not happened. Digging further it was found that when creating a function we add it to the registry and also to the metastore. If the later fails, we do not clean it up from the registry and thus it remains visible after failure. Fixed the same.",
        "duedate": null,
        "environment": null,
        "fixVersions": [{
            "archived": false,
            "id": "12343343",
            "name": "4.0.0",
            "released": false,
            "self": "https://issues.apache.org/jira/rest/api/2/version/12343343"
        }],
        "issuelinks": [],
        "issuetype": {
            "avatarId": 21133,
            "description": "A problem which impairs or prevents the functions of the product.",
            "iconUrl": "https://issues.apache.org/jira/secure/viewavatar?size=xsmall&avatarId=21133&avatarType=issuetype",
            "id": "1",
            "name": "Bug",
            "self": "https://issues.apache.org/jira/rest/api/2/issuetype/1",
            "subtask": false
        },
        "labels": ["pull-request-available"],
        "lastViewed": null,
        "priority": {
            "iconUrl": "https://issues.apache.org/jira/images/icons/priorities/major.svg",
            "id": "3",
            "name": "Major",
            "self": "https://issues.apache.org/jira/rest/api/2/priority/3"
        },
        "progress": {
            "percent": 0,
            "progress": 0,
            "total": 86400
        },
        "project": {
            "avatarUrls": {
                "16x16": "https://issues.apache.org/jira/secure/projectavatar?size=xsmall&pid=12310843&avatarId=11935",
                "24x24": "https://issues.apache.org/jira/secure/projectavatar?size=small&pid=12310843&avatarId=11935",
                "32x32": "https://issues.apache.org/jira/secure/projectavatar?size=medium&pid=12310843&avatarId=11935",
                "48x48": "https://issues.apache.org/jira/secure/projectavatar?pid=12310843&avatarId=11935"
            },
            "id": "12310843",
            "key": "HIVE",
            "name": "Hive",
            "projectCategory": {
                "description": "Scalable Distributed Computing",
                "id": "10292",
                "name": "Hadoop",
                "self": "https://issues.apache.org/jira/rest/api/2/projectCategory/10292"
            },
            "self": "https://issues.apache.org/jira/rest/api/2/project/12310843"
        },
        "reporter": {
            "active": true,
            "avatarUrls": {
                "16x16": "https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=ashutosh.bapat&avatarId=36785",
                "24x24": "https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=ashutosh.bapat&avatarId=36785",
                "32x32": "https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=ashutosh.bapat&avatarId=36785",
                "48x48": "https://issues.apache.org/jira/secure/useravatar?ownerId=ashutosh.bapat&avatarId=36785"
            },
            "displayName": "Ashutosh Bapat",
            "key": "ashutosh.bapat",
            "name": "ashutosh.bapat",
            "self": "https://issues.apache.org/jira/rest/api/2/user?username=ashutosh.bapat",
            "timeZone": "Asia/Kolkata"
        },
        "resolution": {
            "description": "A fix for this issue is checked into the tree and tested.",
            "id": "1",
            "name": "Fixed",
            "self": "https://issues.apache.org/jira/rest/api/2/resolution/1"
        },
        "resolutiondate": "2018-12-03T12:03:35.000+0000",
        "status": {
            "description": "A resolution has been taken, and it is awaiting verification by reporter. From here issues are either reopened, or are closed.",
            "iconUrl": "https://issues.apache.org/jira/images/icons/statuses/resolved.png",
            "id": "5",
            "name": "Resolved",
            "self": "https://issues.apache.org/jira/rest/api/2/status/5",
            "statusCategory": {
                "colorName": "green",
                "id": 3,
                "key": "done",
                "name": "Done",
                "self": "https://issues.apache.org/jira/rest/api/2/statuscategory/3"
            }
        },
        "subtasks": [],
        "summary": "Remove a function from function registry when it can not be added to the metastore when creating it.",
        "timeestimate": 86400,
        "timeoriginalestimate": 86400,
        "timespent": null,
        "updated": "2018-12-04T04:31:14.000+0000",
        "versions": [{
            "archived": false,
            "id": "12343343",
            "name": "4.0.0",
            "released": false,
            "self": "https://issues.apache.org/jira/rest/api/2/version/12343343"
        }],
        "votes": {
            "hasVoted": false,
            "self": "https://issues.apache.org/jira/rest/api/2/issue/HIVE-20953/votes",
            "votes": 0
        },
        "watches": {
            "isWatching": false,
            "self": "https://issues.apache.org/jira/rest/api/2/issue/HIVE-20953/watchers",
            "watchCount": 4
        },
        "workratio": 0
    },
    "id": "13199756",
    "key": "HIVE-20953",
    "self": "https://issues.apache.org/jira/rest/api/2/issue/13199756"
}